xbmc/application/ApplicationPlayer.cpp

```diff
@@ -171,35 +171,34 @@ bool CApplicationPlayer::HasPlayer() const
   return player != nullptr;
 }
 
-int CApplicationPlayer::GetChapter()
+int CApplicationPlayer::GetChapter() const
 {
-  std::shared_ptr<IPlayer> player = GetInternal();
+  const auto player = GetInternal();
```

Comment:
Only thing I'll say is that the use of auto here isn't that great, it's not obvious what the type is and if it's even a pointer (or smart pointer). There is no indication of the type from the method call either.

I would think it would be more readable to keep `const std::shared_ptr<IPlayer> player` but that's just me. We are in the days of modern IDE's so checking the type is as easy as hovering the cursor.

Reply:
changed to explicit type.